### PR TITLE
Move warning and error flags into CI build flags.

### PR DESCRIPTION
**Pull request by GleasonK**
Per Mehdi's comments on #137. It is safer to enforce warnings as errors in an environment that we have full control over.

### CODE REVIEW

**Review by burmako — APPROVED** [review]
LGTM, but let's also wait for a review from Mehdi.

**Review by joker-eph — APPROVED** [review]

**Pull request merged**
